subjects/devops/crud-master/README.md

```diff
@@ -0,0 +1,270 @@
+
+#### Virtual Machines
+
+##### General overview
+
+You will use DropBox and Vagrant to setup three different VMs in order to test the interactions and correctness of responses between your APIs infrastructure.
+
+Vagrant is an open-source software that helps you create and manage virtual machines. With Vagrant, you can create a development environment that is identical to your production environment, which makes it easier to develop, test, and deploy your applications.
+
+Your VMs will be structured as follow:
+
+- `gateway-vm`: This VM will only contain the `api-gateway`.
+- `inventory-vm`: This VM will contain the `inventory-app` API and the database `movies`.
+- `billing-vm`: This VM will contain the `billing-app` API, database `orders` and RabbitMQ.
+
+> Vagrant is designed for development and should not be used in production environments.
+
+##### Environment variables
+
+To simplify the building process it is a common practice to store useful variables in a `.env` file. It will make very easy to modify/update information like URLs, passwords, users and so on.
+
+Those variables will then be used by Vagrant and passed through the different APIs in order to centralize all the credentials.
+
+Your `.env` file should contain the following variables:
+
+```console
+# postgres password
+POSTGRES_PASSWORD=...
+# postgres user for billing database
+BILLING_DB_USER=...
+# postgres password for billing database
+BILLING_DB_PASSWORD=...
+# postgres name for billing database
+BILLING_DB_NAME=...
+# postgres user for inventory database
+INVENTORY_DB_USER=...
+# postgres password for inventory database
+INVENTORY_DB_PASSWORD=...
+# postgres name for inventory database
+INVENTORY_DB_NAME=...
+# rabbitmq user
+RABBITMQ_USER=...
+# rabbitmq passowrd
+RABBITMQ_PASSWORD=...
+# rabbitmq port
+RABBITMQ_PORT=5672
+# rabbitmq billing queue name
+RABBITMQ_BILLING_QUEUE=billing_queue
+# inventory api port
+INV_APP_PORT=3001
+# apigateway api port
+APIGATEWAY_PORT=3000
+```
+
+> For the purpose of this exercise the `.env` file must be included in your repository, though usually in public and production projects the `.env` file is never included in repos to avoid sensitive data leaks.
+
+##### Configuration of the VMs
+
+- You will have a `Vagrantfile` which will create and start the three VMs. It will import the environment variables and pass them through each API.
+- You will have a `script/` directory which will store all the scripts you may want to run in order to install the necessary tools on each VM. Those scripts may also be very useful for setting up the databases.
+
+Your configuration will work properly for the following commands (executed from the root of the project):
+
+- `vagrant up --provider virtualbox`: Starts all the VMs.
+- `vagrant status`: Shows the status for all the VMs.
+- `vagrant ssh <vm-name>`: Will let you access the VM through SSH.
+
+#### Manage Your Node.js applications with PM2
```

Comment:
we need to be sure the student will be able to start and stop services in a streamlined way without relying too much on every different possible implementation. Also students will have very little knowledge at this point of the curriculum and not really a lot of time to come up with a solution.
If it's ok for you I think it will be profitable to lead them into a specific direction on this one.

Let me know, if you still feel it's tmi I'll remove the section, but then the audit need to be changed and becomes very generic for this part

Reply:
It's fine you can keep it